subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,24 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+//printk("%u %u\n", sys_get_be32(NET_TCP_HDR(pkt)->seq), context->tcp->send_ack);
+
+	/* Peer sent us packet we've already seen. Apparently, our ack was lost. */
+	if (sys_get_be32(NET_TCP_HDR(pkt)->seq) < context->tcp->send_ack) {
+printk("Received already seen pkt - ACKing\n");
+
+	set_appdata_values(pkt, IPPROTO_TCP);
```

Comment:
> I thought that sending ACK to some value would automatically acknowledge all the previous values.

Then the explanation of what happens is less clear. Were you able to reproduce the original issue (against archive.ubuntu.com or another site) and confirm that this patch makes a difference? We can try to just resend our current ack value (without trying to "rewind" it back), and see if that still works. I'll try that a bit later unless you beat on on that.